### PR TITLE
Plane: clang fails - cosf cannot be used in constexpr

### PR DESCRIPTION
**Pull request by drnic**
Fixes macos/gcc/clang build error:

```
[483/500] Compiling ArduPlane/quadplane.cpp
../../ArduPlane/tailsitter.cpp:274:29: fatal error: constexpr variable 'c_trans_angle' must be initialized by a constant expression
            constexpr float c_trans_angle = cosf(.125f * M_PI);
                            ^               ~~~~~~~~~~~~~~~~~~
../../ArduPlane/tailsitter.cpp:274:45: note: non-constexpr function 'cosf' cannot be used in a constant expression
            constexpr float c_trans_angle = cosf(.125f * M_PI);
                                            ^
/usr/include/math.h:338:14: note: declared here
extern float cosf(float);
             ^
1 error generated.
```

### CODE REVIEW

**Comment by khancyr** [comment]
That seems a step back ...are you sure that you get a recent clang version ?

**Comment by drnic** [comment]
@khancyr 
```
$ gcc -v
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 10.0.0 (clang-1000.10.44.4)
Target: x86_64-apple-darwin18.2.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**Comment by drnic** [comment]
I'm happy for any solution. I just discovered that `./Tools/autotest/sim_vehicle.py -v ArduPlane` was failing to build on my MacOS and proposed this solution.

**Comment by peterbarker** [comment]
@khancyr The spec doesn't include the trig functions as things that are constexpr, thus this failure.

We may want to move these definitions into an initialisation function - depending on how the compilation works out.  We don't want to be doing trig functions where these currently are.

**Comment by peterbarker** [comment]
@kd0aij for visibility

**Comment by khancyr** [comment]
Well, it failed on Ubuntu with Clang6 and 7 too ... 
A fix could be to static const those, can it ?

**Comment by kd0aij** [comment]
according to godbolt, const is OK with clang7: https://godbolt.org/z/_8vNZC

**Comment by lvale** [comment]
also happens with more recent versions

`$ gcc -v
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 10.0.0 (clang-1000.11.45.5)
Target: x86_64-apple-darwin18.2.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin`

**Comment by peterbarker** [comment]
Marked DevCallTopic so it gets looked at.

Playing with godbolt it is possible to get it to actually call into cosf, but you have to really work at it (`-O0` and clang++)

tridge may want to play it safe and move the definitions elsewhere.

File: ArduPlane/tailsitter.cpp [hunk 1]
```diff
@@ -13,7 +13,7 @@
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 /*
-  control code for tailsitters. Enabled by setting Q_FRAME_CLASS=10
```
Comment (peterbarker) [inline ArduPlane/tailsitter.cpp]: These unrelated whitespace changes should be removed.  They make it harder to trace history.

Comment (drnic) [reply ArduPlane/tailsitter.cpp]: I've removed the whitespace fixes now.

Comment (drnic) [reply ArduPlane/tailsitter.cpp]: CI has passed

**Comment by drnic** [comment]
Since https://github.com/ArduPilot/ardupilot/pull/10901 was merged, I'll close this

**Comment by peterbarker** [comment]
On Wed, 27 Mar 2019, Dr Nic Williams wrote:

> Since #10901 was merged, I'll close this

Thanks for this.  We really need to put clang++ more prominently in our 
testing system as we keep breaking things for the poor MacOSX users...

**Comment by lvale** [comment]
> we keep breaking things for the poor MacOSX users...

Don't worry, we'll keep nagging you @peterbarker :)
